src/wp-includes/class-wp-block.php

```diff
@@ -139,6 +139,34 @@ public function __construct( $block, $available_context = array(), $registry = n
 		$this->block_type = $registry->get_registered( $this->name );
 
 		$this->available_context = $available_context;
+		$this->update_available_context( $block, $available_context );
+	}
+
+	/**
+	 * Updates the available context for the current block and its inner blocks.
+	 *
+	 * @since 6.7.0
+	 *
+	 * Updates the available context for the current block and its inner blocks.
+	 *
+	 * This method updates the context of the current block instance by merging the provided
+	 * `available_context` with the existing context values. It also processes the block's
+	 * inner blocks by passing the updated context to them.
+	 *
+	 * The available context is an array of key-value pairs that represent the context passed
+	 * down from ancestor blocks in the hierarchy. The block instance's context is only updated
+	 * with the values that it consumes as defined in its registered block type (`uses_context`).
+	 * Additionally, any context provided by the block instance itself is passed to its inner blocks.
```

Comment:
NIt: it could help to remove the duplication in the comment:
```suggestion
	/**
	 * Updates the available context for the current block and its inner blocks.
	 *
	 * This method updates the context of the current block instance by merging the provided
	 * `available_context` with the existing context values. It also processes the block's
	 * inner blocks by passing the updated context to them.
	 *
	 * The available context is an array of key-value pairs that represent the context passed
	 * down from ancestor blocks in the hierarchy. The block instance's context is only updated
	 * with the values that it consumes as defined in its registered block type (`uses_context`).
	 * Additionally, any context provided by the block instance itself is passed to its inner blocks.
	 *
	 * @since 6.7.0
```